src/System.Data.Common/src/System/Data/Common/DbDataReaderExtensions.netcoreapp.cs

```diff
@@ -0,0 +1,181 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.ComponentModel;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace System.Data.Common
```

Comment:
I understand this class happens to exist and is located under System.Data.Common, but I always thought that I would only add a using statement for this namespace if I was:
1. Building an ADO.NET provider
2. Consuming ADO.NET APIs in a provider-agnostic way

It may be a bit subjective, but I think System.Data would be a more appropriate namespace for extension methods that apply to all ADO.NET providers. 

I hope that switching to extension methods isn’t going to limit the impact of this change on usability. 

Thoughts?

Reply:
Seems valid. The problem with moving them to a higher namespace is that it'll mean that the parent namespace references a child namespace because of the this parameter. So while it's tempting to suggest moving this up to a new `System.Data.DataReaderExtensions` i don't think it'd pass review.  We can't move `DbDataReader` without a netstandard move or duplication which again i think would fare badly in review.

Reply:
A comment or two from my side... it's true that System.Data.Common is pretty rarely used, and it does seem much better for discoverability to have these extension methods in System.Data (which everyone already uses).

> The problem with moving them to a higher namespace is that it'll mean that the parent namespace references a child namespace because of the this parameter.

It may be a bit odd, but I'm not sure there's any actual specific problem with having an extension method in System.Data refer to something in System.Data.Common. If it's a matter of coding practices/standards perhaps an exception can be made for the ADO.NET case.

A second issue is that there's already the other DbDataReaderExtensions (with GetColumnSchema()) living in System.Data.Common. We can add a new DbDataReaderExtensions class under System.Data with the new overloads, and then we'd have two DbDataReaderExtensions (the old one needs to stay for backwards compat. This seems messy but I'm not sure it has any actual downsides.

Note that this seems to be the only extensions class in ADO.NET for the moment.

Reply:
+1 for System.Data namespace. This approach has generally worked very well in the past for discoverability of extension methods.

Reply:
It would be better to have the extensions in System.Data as @divega said, if you think it'll be accepted to have them on a class in that namespace then it's worth asking for.

Reply:
Did we reach a final conclusion of this? Shall I move the extensions to `System.Data` namespace as a new `DbDataReaderExtensions` class?

Reply:
Please, do it. I will follow up with API reviewers on whether they have concerns with having two extension method sponsor classes with same name in two different namespaces.

I will also create a new issue about “relocating” (or copying and deprecating) GetSchema to the new class, but you don’t need to worry about that on this PR.

Cc @terrajobst.

Reply:
I added `DataReaderExtensions` under `System.Data` namespace.